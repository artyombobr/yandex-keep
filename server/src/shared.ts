export interface NotesProps {
  data: DataEntity;
}

export interface DataEntity {
  tags: TagsEntity[];
  colors: ColorsEntity[];
  notes: NotesEntity[];
}

export interface TagsEntity {
  id: number;
  tag: string;
}

export interface ColorsEntity {
  id: number;
  color: string;
}

export interface NotesEntity {
  type: string;
  title?: string;
  tags?: number[];
  color?: number;
  items?: ItemsEntity[];
  size?: string;
  created: number;
  text?: string;
  attachments?: AttachmentsEntity[];
  reminder?: number;
  url?: string;
  archive?: boolean;
}

export interface ItemsEntity {
  text: string;
  checked: boolean;
}

export interface AttachmentsEntity {
  type: string;
  url: string;
}
