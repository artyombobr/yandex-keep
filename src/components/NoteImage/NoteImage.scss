.note__image img {
  width: 100%;
  border-radius: 16px;
  margin-bottom: 12px;
}
