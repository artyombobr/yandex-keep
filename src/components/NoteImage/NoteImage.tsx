import * as React from 'react';
import NoteFooter from '../NoteFooter/NoteFooter';
import setBackground from '../../helpers/setBackground';

function NoteImage(props: any) {
  const { note, colors, isEdit, setEdit } = props;
  return (
    <div
      className="note__main note__main_image"
      style={colors && setBackground(colors, note)}
    >
      <div className="note__image">
        <picture>
          <source
            media="(max-width: 320px)"
            srcSet="https://via.placeholder.com/320"
          />
          <source
            media="(max-width: 465px)"
            srcSet="https://via.placeholder.com/465"
          />
          <source
            media="(max-width: 650px)"
            srcSet="https://via.placeholder.com/650"
          />
          <img src="https://via.placeholder.com/1020" alt="Изображение" />
        </picture>
      </div>
      <NoteFooter note={note} isEdit={isEdit} setEdit={setEdit} />
    </div>
  );
}

export default NoteImage;
