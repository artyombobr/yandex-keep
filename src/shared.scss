$small: 375px;
$medium: 769px;
$large: 1366px;

@function strip-unit($number) {
  @return $number / ($number * 0 + 1);
}

@function calc-size($f-min, $f-max, $v-min, $v-max) {
  $f-min: strip-unit($f-min);
  $f-max: strip-unit($f-max);
  $v-min: strip-unit($v-min);
  $v-max: strip-unit($v-max);
  $k: ($f-max - $f-min)/($v-max - $v-min);
  $b: $f-min - $k * $v-min;
  $b: $b * 1px;
  @return calc( #{$k} * 100vw + #{$b} );
}

@mixin fontsize($f-min, $f-max, $l-min, $l-max, $w-min, $w-max) {
  font-size: $f-min;
  @if $l-min !=0 {
    line-height: $l-min;
  }
  @media (min-width: $w-min+1) {
    font-size: calc-size($f-min, $f-max, $w-min, $w-max);
    @if $l-min !=0 {
      line-height: calc-size($l-min, $l-max, $w-min, $w-max);
    }
  }
  @media (min-width: $w-max) {
    font-size: $f-max;
    @if $l-max !=0 {
      line-height: $l-max;
    }
  }
}
