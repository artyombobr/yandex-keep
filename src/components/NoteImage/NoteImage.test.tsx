import React from 'react';
import { shallow } from 'enzyme';
import NoteImage from './NoteImage';
import { ColorsEntity } from '../../shared';

describe('NoteImage', () => {
  const note = {
    id: 1,
    type: 'text',
    text: 'Купить чайник с Bluetooth\nВидел у Xiaomi на днях',
    color: 0,
    tags: [0],
    size: 's',
    created: 1551593220000,
  };
  const colors: ColorsEntity[] = [
    {
      id: 0,
      color: '#E84747',
    },
  ];

  it('renders correctly', () => {
    const component = shallow(<NoteImage note={note} colors={colors} />);
    expect(component).toMatchSnapshot();
  });
});
