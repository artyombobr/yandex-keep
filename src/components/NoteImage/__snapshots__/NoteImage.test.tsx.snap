// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteImage renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoteImage
    colors={
      Array [
        Object {
          "color": "#E84747",
          "id": 0,
        },
      ]
    }
    note={
      Object {
        "color": 0,
        "created": 1551593220000,
        "id": 1,
        "size": "s",
        "tags": Array [
          0,
        ],
        "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
        "type": "text",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="note__image"
        >
          <picture>
            <source
              media="(max-width: 320px)"
              srcSet="https://via.placeholder.com/320"
            />
            <source
              media="(max-width: 465px)"
              srcSet="https://via.placeholder.com/465"
            />
            <source
              media="(max-width: 650px)"
              srcSet="https://via.placeholder.com/650"
            />
            <img
              alt="Изображение"
              src="https://via.placeholder.com/1020"
            />
          </picture>
        </div>,
        <UNDEFINED
          note={
            Object {
              "color": 0,
              "created": 1551593220000,
              "id": 1,
              "size": "s",
              "tags": Array [
                0,
              ],
              "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
              "type": "text",
            }
          }
        />,
      ],
      "className": "note__main note__main_image",
      "style": Object {
        "backgroundColor": "rgb(246,181,181)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <picture>
            <source
              media="(max-width: 320px)"
              srcSet="https://via.placeholder.com/320"
            />
            <source
              media="(max-width: 465px)"
              srcSet="https://via.placeholder.com/465"
            />
            <source
              media="(max-width: 650px)"
              srcSet="https://via.placeholder.com/650"
            />
            <img
              alt="Изображение"
              src="https://via.placeholder.com/1020"
            />
          </picture>,
          "className": "note__image",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <source
                media="(max-width: 320px)"
                srcSet="https://via.placeholder.com/320"
              />,
              <source
                media="(max-width: 465px)"
                srcSet="https://via.placeholder.com/465"
              />,
              <source
                media="(max-width: 650px)"
                srcSet="https://via.placeholder.com/650"
              />,
              <img
                alt="Изображение"
                src="https://via.placeholder.com/1020"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "media": "(max-width: 320px)",
                "srcSet": "https://via.placeholder.com/320",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "media": "(max-width: 465px)",
                "srcSet": "https://via.placeholder.com/465",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "media": "(max-width: 650px)",
                "srcSet": "https://via.placeholder.com/650",
              },
              "ref": null,
              "rendered": null,
              "type": "source",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Изображение",
                "src": "https://via.placeholder.com/1020",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "picture",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "note": Object {
            "color": 0,
            "created": 1551593220000,
            "id": 1,
            "size": "s",
            "tags": Array [
              0,
            ],
            "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
            "type": "text",
          },
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(NoteFooter)",
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="note__image"
          >
            <picture>
              <source
                media="(max-width: 320px)"
                srcSet="https://via.placeholder.com/320"
              />
              <source
                media="(max-width: 465px)"
                srcSet="https://via.placeholder.com/465"
              />
              <source
                media="(max-width: 650px)"
                srcSet="https://via.placeholder.com/650"
              />
              <img
                alt="Изображение"
                src="https://via.placeholder.com/1020"
              />
            </picture>
          </div>,
          <UNDEFINED
            note={
              Object {
                "color": 0,
                "created": 1551593220000,
                "id": 1,
                "size": "s",
                "tags": Array [
                  0,
                ],
                "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
                "type": "text",
              }
            }
          />,
        ],
        "className": "note__main note__main_image",
        "style": Object {
          "backgroundColor": "rgb(246,181,181)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <picture>
              <source
                media="(max-width: 320px)"
                srcSet="https://via.placeholder.com/320"
              />
              <source
                media="(max-width: 465px)"
                srcSet="https://via.placeholder.com/465"
              />
              <source
                media="(max-width: 650px)"
                srcSet="https://via.placeholder.com/650"
              />
              <img
                alt="Изображение"
                src="https://via.placeholder.com/1020"
              />
            </picture>,
            "className": "note__image",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <source
                  media="(max-width: 320px)"
                  srcSet="https://via.placeholder.com/320"
                />,
                <source
                  media="(max-width: 465px)"
                  srcSet="https://via.placeholder.com/465"
                />,
                <source
                  media="(max-width: 650px)"
                  srcSet="https://via.placeholder.com/650"
                />,
                <img
                  alt="Изображение"
                  src="https://via.placeholder.com/1020"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "media": "(max-width: 320px)",
                  "srcSet": "https://via.placeholder.com/320",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "media": "(max-width: 465px)",
                  "srcSet": "https://via.placeholder.com/465",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "media": "(max-width: 650px)",
                  "srcSet": "https://via.placeholder.com/650",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Изображение",
                  "src": "https://via.placeholder.com/1020",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "picture",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "note": Object {
              "color": 0,
              "created": 1551593220000,
              "id": 1,
              "size": "s",
              "tags": Array [
                0,
              ],
              "text": "Купить чайник с Bluetooth
Видел у Xiaomi на днях",
              "type": "text",
            },
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(NoteFooter)",
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
